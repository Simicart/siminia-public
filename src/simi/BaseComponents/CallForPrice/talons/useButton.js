import { useMemo, useCallback } from 'react';
import { getHidePriceEnable } from '../utils';
import { useAppContext } from '@magento/peregrine/lib/context/app';

export const useButton = (props = {}) => {
    const { product } = props;

    const { advancedhideprice, sku } = product || {};
    const [, { toggleDrawer }] = useAppContext();
    const hidePriceEnable = getHidePriceEnable();

    const isHidePrice = useMemo(() => {
        return (
            hidePriceEnable &&
            advancedhideprice &&
            advancedhideprice.advancedhideprice_type === 'hideprice'
        );
    }, [hidePriceEnable, advancedhideprice]);

    const isCallForPrice = useMemo(() => {
        return (
            hidePriceEnable &&
            advancedhideprice &&
            advancedhideprice.advancedhideprice_type === 'callforprice'
        );
    }, [hidePriceEnable, advancedhideprice]);

    const text = useMemo(() => {
        return advancedhideprice?.advancedhideprice_text;
    }, [advancedhideprice]);

    const handleOpen = useCallback(() => {
        toggleDrawer('hidePrice.form.' + sku);
    }, [toggleDrawer]);

    return {
        isHidePrice,
        isCallForPrice,
        text,
        productSku: sku,
        handleOpen
    };
};
