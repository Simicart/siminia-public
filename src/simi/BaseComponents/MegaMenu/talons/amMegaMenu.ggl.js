import { gql } from '@apollo/client';

export const GET_AM_MEGAMENU_CONFIG = gql`
  query amMegaMenuConfig {
    storeConfig {
      store_code
      root_category_uid
      ammegamenu_general_enabled
      ammegamenu_general_sticky
      ammegamenu_general_hamburger_enabled
      ammegamenu_general_mobile_template
      ammegamenu_general_show_icons
      ammegamenu_color_color_template
      ammegamenu_color_main_menu_background
      ammegamenu_color_main_menu_background_hover
      ammegamenu_color_main_menu_text
      ammegamenu_color_main_menu_text_hover
      ammegamenu_color_submenu_background_color
      ammegamenu_color_submenu_background_image
      ammegamenu_color_submenu_text
      ammegamenu_color_submenu_text_hover
      ammegamenu_color_current_category_color
      ammegamenu_color_toggle_icon_color
    }
  }
`;

export const GET_AM_MEGAMENU_ITEMS = gql`
  query amMegaMenuAll {
    amMegaMenuAll {
      items {
        name
        id
        uid
        url
        parent_id
        parent_uid
        width
        column_count
        content
        has_active
        is_active
        is_category
        is_parent_active
        width_value
        status
        label
        label_text_color
        label_background_color
        icon
        submenu_type
        subcategories_position
      }
    }
  }
`;

export default {
  getAmMegaMenuConfig: GET_AM_MEGAMENU_CONFIG,
  getAmMegaMenuItems: GET_AM_MEGAMENU_ITEMS
};
