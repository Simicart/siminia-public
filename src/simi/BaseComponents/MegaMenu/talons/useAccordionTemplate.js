import { useMemo } from 'react';
import { useAmMegaMenuContext } from '../context';
import { buildTree } from '../utils';

const CATEGORY_IDX = 'category-node-';

export const useAccordionTemplate = () => {
  const { allItems: items, isMobile, config } = useAmMegaMenuContext();
  const { root_category_uid: rootCategoryId } = config;

  const menuTree = useMemo(() => {
    if (!Array.isArray(items) || !items.length) {
      return null;
    }

    const itemList = isMobile
      ? items
      : items.filter(item => item.id.includes(CATEGORY_IDX));

    return buildTree(itemList, rootCategoryId);
  }, [items, isMobile, rootCategoryId]);

  return {
    menuTree
  };
};
