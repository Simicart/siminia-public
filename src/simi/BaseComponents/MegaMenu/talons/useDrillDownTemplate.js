import { useAmMegaMenuContext } from '../context';
import { useEffect, useMemo } from 'react';
import { buildTree, mapItems } from '../utils';
import { useLazyQuery } from '@apollo/client';
import DEFAULT_OPERATIONS from '@magento/peregrine/lib/talons/CategoryTree/categoryTree.gql';
import mergeOperations from '@magento/peregrine/lib/util/shallowMerge';

export const useDrillDownTemplate = props => {
  const { categoryId, updateCategories } = props;
  const { allItems, config, isShowIcons } = useAmMegaMenuContext();

  const operations = mergeOperations(DEFAULT_OPERATIONS, props.operations);
  const { getNavigationMenuQuery } = operations;

  const [runQuery, queryResult] = useLazyQuery(getNavigationMenuQuery, {
    fetchPolicy: 'no-cache',
    // nextFetchPolicy: 'cache-first'
  });
  const { data } = queryResult;

  // fetch categories
  useEffect(() => {
    if (categoryId != null) {
      runQuery({ variables: { id: categoryId } });
    }
  }, [categoryId, runQuery]);

  // update redux with fetched categories
  useEffect(() => {
    if (data && data.category) {
      updateCategories(data.category);
    }
  }, [data, updateCategories]);

  const { root_category_id: rootCategoryId } = config;
  const isTopLevel = categoryId === rootCategoryId;

  const categories = useMemo(() => {
    if (!Array.isArray(allItems)) {
      return null;
    }

    const tree = buildTree(mapItems(allItems), true);

    if (isTopLevel) {
      return [...tree.values()].filter(({ parent_id }) => !parent_id);
    }

    const category = tree.get(categoryId);
    // const { subCategories, name } = category;
    const subCategories = category && category.subCategories ? category.subCategories : ''
    const name = category && category.name ? category.name : ''

    return [
      { ...category, name: `All ${name}`, subCategories: null },
      ...subCategories
    ];
  }, [allItems, categoryId, isTopLevel]);

  return {
    categories,
    isShowIcons,
    isTopLevel
  };
};
