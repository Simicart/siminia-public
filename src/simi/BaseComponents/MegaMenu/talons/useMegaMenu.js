import { useCallback, useMemo, useState } from 'react';
import { useQuery } from '@apollo/client';
import { useWindowSize } from '@magento/peregrine';
import mergeOperations from '@magento/peregrine/lib/util/shallowMerge';
import { mapConfig } from '../utils';

import DEFAULT_OPERATIONS from './amMegaMenu.ggl';

export const useMegaMenu = (props = {}) => {
  const windowSize = useWindowSize();
  const [openCategoryIds, setOpenCategoryIds] = useState(new Set([]));

  const operations = mergeOperations(DEFAULT_OPERATIONS, props.operations);
  const { getAmMegaMenuConfig, getAmMegaMenuItems } = operations;

  const { data: storeConfigData } = useQuery(getAmMegaMenuConfig, {
    fetchPolicy: 'cache-and-network',
    nextFetchPolicy: 'cache-first'
  });

  const { loading, error, data } = useQuery(getAmMegaMenuItems, {
    fetchPolicy: 'cache-and-network',
    nextFetchPolicy: 'cache-first'
  });

  const { amMegaMenuAll } = data || {};
  const { items } = amMegaMenuAll || {};

  const config = useMemo(() => {
    const { storeConfig } = storeConfigData || {};

    return storeConfig ? mapConfig({ ...storeConfig }) : {};
  }, [storeConfigData]);

  const {
    ammegamenu_general_enabled,
    ammegamenu_general_show_icons,
    ammegamenu_general_mobile_template,
    ammegamenu_color_color_template,
    ammegamenu_color_main_menu_background,
    ammegamenu_color_toggle_icon_color,
    ammegamenu_color_main_menu_text,
    ammegamenu_general_hamburger_enabled,
    ammegamenu_general_sticky,
    ammegamenu_color_current_category_color,
    ammegamenu_color_submenu_text
  } = config;

  const isMobile = windowSize.innerWidth <= 1024;
  const isEnabledMegaMenu = ammegamenu_general_enabled === true;

  const isShowIcons =
    ammegamenu_general_show_icons === 'desktopAndMobile' ||
    (!isMobile && ammegamenu_general_show_icons === 'desktop') ||
    (isMobile && ammegamenu_general_show_icons === 'mobile');

  const isShowHamburgerBtn = isMobile || ammegamenu_general_hamburger_enabled;

  const amRootStyle =
    isEnabledMegaMenu && ammegamenu_color_color_template !== 'blank'
      ? {
          '--am-mega-menu-root-bg': ammegamenu_color_main_menu_background,
          '--am-mega-menu-toggle-color': ammegamenu_color_toggle_icon_color,
          '--am-mega-menu-text-color': ammegamenu_color_main_menu_text,
          '--am-mega-menu-position': ammegamenu_general_sticky
            ? 'sticky'
            : 'relative',
          '--am-mega-menu-toggle-display': !isShowHamburgerBtn
            ? 'none'
            : undefined,
          '--am-mega-menu-active-color': ammegamenu_color_current_category_color,
          '--am-mega-menu-submenu-text-color': ammegamenu_color_submenu_text
        }
      : null;

  const toggleAction = useCallback(
    catId => {
      setOpenCategoryIds(prevOpenCategoryIds => {
        const nextOpenCategoryIds = new Set(prevOpenCategoryIds);
        const rootItems =
          items &&
          items.filter(({ parent_id }) => !parent_id).map(({ id }) => id);

        if (!prevOpenCategoryIds.has(catId)) {
          if (!isMobile || rootItems.includes(catId)) {
            nextOpenCategoryIds.clear();
          }

          nextOpenCategoryIds.add(catId);
        } else {
          nextOpenCategoryIds.delete(catId);
        }

        return nextOpenCategoryIds;
      });
    },
    [setOpenCategoryIds, isMobile, items]
  );

  return {
    loading,
    error,
    isEnabledMegaMenu,
    isMobile,
    hamburgerView: ammegamenu_general_mobile_template,
    isShowIcons,
    config,
    allItems: items,
    amRootStyle,
    openCategoryIds,
    toggleAction
  };
};
