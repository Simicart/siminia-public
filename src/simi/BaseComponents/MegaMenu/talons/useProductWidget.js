import GET_PRODUCT_WIDGET from '../queries/getProductWidget.graphql';
import { useQuery } from '@apollo/client';

export const useProductWidget = props => {
  const { widgetId } = props;

  const { loading, error, data } = useQuery(GET_PRODUCT_WIDGET, {
    variables: {
      widgetId
    }
  });

  const { amMegaMenuWidget } = data || {};

  return {
    ...amMegaMenuWidget,
    loading,
    error
  };
};
