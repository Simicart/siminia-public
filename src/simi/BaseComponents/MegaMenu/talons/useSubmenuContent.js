import { useCallback, useEffect, useState } from 'react';

export const useSubMenuContent = props => {
  const { id } = props;
  const [activeId, setActiveId] = useState(id);

  const onHover = useCallback(_id => setActiveId(_id), [setActiveId]);

  useEffect(() => {
    return () => setActiveId(id);
  }, [setActiveId, id]);

  return {
    activeId,
    onHover
  };
};
