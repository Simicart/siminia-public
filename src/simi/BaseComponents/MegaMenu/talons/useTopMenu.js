import { useMemo } from 'react';
import { useAmMegaMenuContext } from '../context';
import { buildTree } from '../utils';

const CATEGORY_IDX = 'category-node-';

export const useTopMenu = () => {
  const {
    allItems: items,
    config,
    isEnabledMegaMenu,
    isMobile
  } = useAmMegaMenuContext() || {};

  const menuTree = useMemo(() => {
    const { ammegamenu_general_hamburger_enabled: isHamburger } = config || {};

    if (!Array.isArray(items) || !items.length) {
      return null;
    }

    const itemList = isHamburger
      ? items.filter(item => !item.id.includes(CATEGORY_IDX))
      : items;

    return buildTree(itemList);
  }, [items, config]);

  const isEnabled = !isMobile && isEnabledMegaMenu;

  return {
    config,
    menuTree,
    isEnabled
  };
};
