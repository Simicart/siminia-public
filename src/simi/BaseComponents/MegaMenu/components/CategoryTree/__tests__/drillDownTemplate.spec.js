import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import DrillDownTemplate from '../drillDownTemplate';
import { useDrillDownTemplate } from '../../../talons/useDrillDownTemplate';

jest.mock('../../MenuTree/menuLink', () => 'MenuLink');
jest.mock('../../../talons/useDrillDownTemplate');

const defaultProps = {
  categoryId: 2,
  onNavigate: jest.fn().mockName('onNavigate'),
  setCategoryId: jest.fn().mockName('setCategoryId'),
  updateCategories: jest.fn().mockName('updateCategories')
};

const items = [
  {
    name: 'Bottoms',
    id: 'category-node-11',
    url: 'venia-bottoms.html',
    parent_id: null,
    __typename: 'MenuCategoryItem'
  },
  {
    name: 'Pants & Shorts',
    id: 'category-node-12',
    url: 'venia-bottoms/venia-pants.html',
    parent_id: 'category-node-11'
  }
];

test('renders without data', () => {
  useDrillDownTemplate.mockReturnValue({});

  const tree = createTestInstance(<DrillDownTemplate {...defaultProps} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders correct tree', () => {
  useDrillDownTemplate.mockReturnValue({ categories: items });

  const tree = createTestInstance(<DrillDownTemplate {...defaultProps} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
