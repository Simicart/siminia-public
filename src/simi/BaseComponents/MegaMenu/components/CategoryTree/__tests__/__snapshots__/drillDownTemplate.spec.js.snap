// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct tree 1`] = `
<nav>
  <ul>
    <li>
      <MenuLink
        __typename="MenuCategoryItem"
        classes={
          Object {
            "toggle": undefined,
          }
        }
        id="category-node-11"
        name="Bottoms"
        onNavigate={[MockFunction onNavigate]}
        parent_id={null}
        toggleAction={[MockFunction setCategoryId]}
        url="venia-bottoms.html"
        view="hamburger"
      />
    </li>
    <li>
      <MenuLink
        classes={
          Object {
            "toggle": undefined,
          }
        }
        id="category-node-12"
        name="Pants & Shorts"
        onNavigate={[MockFunction onNavigate]}
        parent_id="category-node-11"
        toggleAction={[MockFunction setCategoryId]}
        url="venia-bottoms/venia-pants.html"
        view="hamburger"
      />
    </li>
  </ul>
</nav>
`;

exports[`renders without data 1`] = `null`;
